### ISSUE TITLE
How should we allow for testing flag overrides by testers/QA?

### ISSUE THREAD

**Issue by motin** [mozilla/shield-studies-addon-utils#212]
To be able to provide [proper instructions to QA](https://github.com/motin/taar-experiment-v3-shield-study/blob/develop/docs/TESTPLAN.md#install-the-add-on-and-enroll-in-the-study), without requiring a way too powerful and general browser.prefs API, I added a browser.testingOverrides api in https://github.com/motin/taar-experiment-v3-shield-study and subsequently ported this to utils in https://github.com/mozilla/shield-studies-addon-utils/pull/210 

gregglind commented in https://github.com/mozilla/shield-studies-addon-utils/pull/210#discussion_r197116518:

> People are leaving in the magic prefs even during deployment. #211 . I am not convinced that doing this as a separate api is better.
> 
> -1
> 
> I would prefer a discussion bug to get the list, name the prefs, and discuss how to get them into the product.
> 
> See #208
> 
> Should they have addon specific names?
> Should they persist?

gregglind commented in https://github.com/mozilla/shield-studies-addon-utils/pull/210#discussion_r197117717:

> My suggestion is to have this be json rather than 20 different prefs. People might have opinions on this though.
> 
> It's also a smaller api : "getTestingOverrides(). => {firstRunTs, blah blah}"

**Comment by motin** [comment]
> I would prefer a discussion bug to get the list, name the prefs, and discuss how to get them into the product.

The list:
 * variationName - to be able to test specific variations
 * firstRunTimestamp - to be able to test the expiration event
 * expired - to be able to test the behavior of an already expired study
 * studyType - to be able to test add-ons with different studyType configurations (more explicitly, being able to test a Pioneer study using the shield data processing pipeline to confirm that the data ends up as expected, without having to deal with the sometimes immense lag of the Pioneer data pipeline)

Suggested names of the prefs:
 * `extensions.${widgetId}.test.variationName`
 * `extensions.${widgetId}.test.firstRunTimestamp`
 * `extensions.${widgetId}.test.expired`
 * `extensions.${widgetId}.test.studyType`

Suggestion on how to get them into the product:
```
  studySetup.testing = {
    variationName: await browser.testingOverrides.getVariationNameOverride(),
    firstRunTimestamp: await browser.testingOverrides.getFirstRunTimestampOverride(),
    expired: await browser.testingOverrides.getExpiredOverride(),
  };
  // internal testing override necessary to be able to test all study types
  studySetup.studyType = await browser.internalTestingOverrides.getStudyTypeOverride();
```

> My suggestion is to have this be json rather than 20 different prefs. People might have opinions on this though.
It's also a smaller api : "getTestingOverrides(). => {firstRunTs, blah blah}"

I don't like the idea of having to instruct QA or add-on developers to fill or edit a preference with JSON contents. It seems more fragile. If only to get a simpler API, a similar effect can be achieved by letting all preferences use the same preference branch (`extensions.${widgetId}.test.*`) and then still expose a method that fetches all preferences in that branch and return it as JSON (`"getTestingOverrides(). => {firstRunTs, blah blah}"`)

**Comment by motin** [comment]
This was addressed in #259
